filters/json.js

```diff
@@ -0,0 +1,62 @@
+// i18next j4 json format defined at https://www.i18next.com/misc/json-format
+import flatten from "flat";
+
+export class JsonFilter {
+    constructor(params) {
+        this.enableARBAnnotations = params?.enableARBAnnotations || false;
+        this.enablePluralSuffixes = params?.enablePluralSuffixes || false;
+        this.emitComments = params?.emitComments || false;
+    }
+
+    async parseResource({ resource }) {
+        var segments = [];
+        var notes = {};
+        for (const [key, value] of Object.entries(flatten(resource))) {
+            if (key.indexOf("@") === -1) {
+                var seg = { sid: key, str: value };
+                notes[key] && (seg.notes = notes[key]);
+                this.enablePluralSuffixes &&
+                    ["_one", "_other", "_zero", "_two", "_few", "_many"].some((plural) => key.endsWith(plural)) &&
+                    (seg.isSuffixPluralized = true);
+                segments.push(seg);
+            } else {
+                if (this.enableARBAnnotations && key.endsWith(".description")) {
+                    const sid = key
+                        .replace("@", "")
+                        .replace(".description", "");
+                    const seg = segments.find((e) => e.sid === sid);
```

Comment:
pretty convoluted and error-prone to do the same thing in 2 completely different code paths. isn't it easier to split regular keys and annotations first (e.g. with `filter()`) and then do a single pass?

Reply:
case in point. both and `parseResource` and `generateTranslatedResource` have a bug in that they're ignoring keys that contain `@` even if `enableARBAnnotations` is false.

Reply:
i wasn't sure what to do in the case of keys with `@` and `enableARBAnnotations` is false.  changed the logic as you suggested.